tests/test_cases/socketselect.py

```diff
@@ -0,0 +1,27 @@
+#!/bin/python2
+# Checks if socketselect.c runs the same natively as it does within Lind
+import sys
+import string
+
+lind_results = sys.argv[1]
+native_results = sys.argv[2]
+
+lind_result_split = lind_results.split('\n')
+native_result_split = native_results.split('\n')
+
+# Deterministic line comparisons
+if len(lind_result_split) != len(native_result_split):
+    print "Mismatched number of lines!\n"
+    print "Potential issue for select: If the test is run in quick successions, there's a possibility that the select has not been closed."
+    exit(-1)
+
+resultLen = len(lind_result_split)
+for line in range(0,resultLen):
+    if lind_result_split[line].translate(None, string.digits+'-') != native_result_split[line].translate(None, string.digits+'-'):
+        print "Nondeterministic lines of an unrecognized format!"
+        exit(-1)
+
+# Hello received check
+if lind_result_split[7] != native_result_split[7]:
```

Comment:
is this always deterministically at this array index? I thought there could be print ordering differences.

Reply:
AFAIK; print order differences happen with fork. But to make sure, I'll sort and do the check.

Reply:
This is threaded so it should potentially have the same issues.